test/Mvc.Api.Analyzers.Test/TestFiles/AddResponseTypeAttributeCodeFixProviderIntegrationTest/CodeFixAddsResponseTypeWhenDifferentFromErrorType.Input.cs

```diff
@@ -0,0 +1,26 @@
+﻿namespace Microsoft.AspNetCore.Mvc.Api.Analyzers._INPUT_
+{
+    [ProducesErrorResponseType(typeof(CodeFixAddsResponseTypeWhenDifferentErrorModel))]
+    [ApiController]
+    [Route("[controller]/[action]")]
+    public class CodeFixAddsResponseTypeWhenDifferentFromErrorType : ControllerBase
+    {
+        public IActionResult GetItem(int id)
+        {
+            if (id == 0)
+            {
+                return NotFound(new CodeFixAddsResponseTypeWhenDifferentErrorModel());
+            }
+
+            if (id == 1)
+            {
+                var validationProblemDetails = new ValidationProblemDetails(ModelState);
+                return BadRequest(validationProblemDetails);
```

Comment:
This test tells the gist of what's getting fixed in this PR. The remainder of it is some refactoring, code cleanup, and tests.